gooddata-sdk/gooddata_sdk/catalog/data_source/declarative_model/data_source.py

```diff
@@ -138,6 +164,28 @@ def to_test_request(
             kwargs["username"] = self.username
         return TestDefinitionRequest(type=self.type, url=self.url, **kwargs)
 
+    def store_to_disk(self, data_sources_folder: Path) -> None:
+        file_path = data_sources_folder / f"{self.id}.yaml"
+        data_source_folder = data_sources_folder / self.id
+        create_directory(data_source_folder)
+        data_source_dict = self.to_api(include_nested_structures=False).to_dict(camel_case=True)
+
+        with open(file_path, "w+", encoding="utf-8") as f:
```

Comment:
Why `w+`? Why it is necessary to open file with `rw` access? I suppose it is this line in documentation: `Modes 'w+' and 'w+b' open and truncate the file`. Is it mean that when I use only `w` and file already exists, it fails?

Reply:
mode fixed + reading/writing layout files unified into utils methods.